src/zenml/integrations/integration.py

```diff
@@ -82,6 +83,11 @@ def check_installation(cls) -> bool:
             )
             return False
 
+    @classmethod
+    def check_system(cls) -> None:
+        """Abstract method to check the system for the integration."""
+        pass
```

Comment:
I was thinking a bit about how we could simplify the code that subclasses need to define. How about we design the method like so:
```suggestion
    @classmethod
    def define_platform_specific_requirements(cls, platform: str) -> List[str]:
        return []
```

and then do the following in `IntegrationMeta.__new__`:

```
platform_specific_requirements = cls.define_platform_specific_requirements(platform.system)
cls.REQUIREMENTS = cls.REQUIREMENTS + platform_specific_requirements
```

Then integration subclasses would neither need `platform.system()` nor override `cls.REQUIREMENTS` every time, which would make the implementation a bit easier perhaps